raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -188,6 +190,11 @@ private void updateReplicaEndOffset(LeaderState state, int replicaId, long endOf
         }
     }
 
+    // For testing only
+    QuorumState quorumState() {
```

Comment:
I'm on the fence about exposing this. Previous test cases validate state through the `QuorumStateStore`. It's true that exposing this allows for deeper validation in some cases, but it also makes the test cases more dependent on the internal implementation rather than focusing on inputs and outputs.